test/Test_Synthesizer.ml

```diff
@@ -39,8 +39,74 @@ let y_MINUS_x_MINUS_z_LE_x () =
+                          ; (Type.INT, Type.INT, [(Value.Int 0 , Value.Int 0)], Value.Int 30) |])
+             ; [| Value.Int 1 ; Value.Int 2 ; Value.Int 3 |] ];
+    outputs = [| Value.Int 10 ; Value.Int 20 ; Value.Int 30 |];
+    constants = []
+  } in Alcotest.(check string) "identical" "(select a k)" result.string
+
+let store () =
+let open Synthesizer in
+let result = solve ~config:{ Config.default with logic = Logic.of_string "ALIA" } {
+  arg_names = [ "a" ; "k" ; "v"];
+  inputs = [ (Array.map ~f:(fun (a,b,c,d) -> Value.Array (a,b,c,d))
+                        [| (Type.INT, Type.INT, [ ],
+                                                Value.Int 1)
+                        ; (Type.INT, Type.INT, [ ],
+                                                Value.Int 0)
+                        ; (Type.INT, Type.INT, [], Value.Int 30) |])
+            ; [| Value.Int 1 ; Value.Int 2 ; Value.Int 3 |]
+            ; [| Value.Int 20 ; Value.Int 40 ; Value.Int 6 |]
+            ];
+  outputs = [| Value.Array((Type.INT, Type.INT, [ (Value.Int 1, Value.Int 20) ],
+                                                  Value.Int 1))
+              ; Value.Array((Type.INT, Type.INT, [ (Value.Int 2, Value.Int 40)],
+                                                   Value.Int 0))
+              ; Value.Array((Type.INT, Type.INT, [ (Value.Int 3 , Value.Int 6)], 
+                                                    Value.Int 30))|];
+  constants = []
+} in Alcotest.(check string) "identical" "(store a k v)" result.string
+
+let two_store () =
+  let open Synthesizer in
+  let result = solve ~config:{ Config.default with logic = Logic.of_string "ALIA" } {
+    arg_names = [ "a" ; "k" ; "v"];
+    inputs = [ (Array.map ~f:(fun (a,b,c,d) -> Value.Array (a,b,c,d))
+                          [| (Type.INT, Type.INT, [ (Value.Int 3, Value.Int 20);],
+                                                  Value.Int 1)
+                          ; (Type.INT, Type.INT, [ (Value.Int 6, Value.Int 20);],
+                                                  Value.Int 0)
+                          ; (Type.INT, Type.INT, [(Value.Int 1, Value.Int 20);], Value.Int 30) |])
+              ; [| Value.Int 1 ; Value.Int 2 ; Value.Int 3 |]
+              ; [| Value.Int 20 ; Value.Int 40 ; Value.Int 6 |]
+              ];
+    outputs = [| Value.Array((Type.INT, Type.INT, [ (Value.Int 1, Value.Int 20) ;(Value.Int 3, Value.Int 20)],
+                                                    Value.Int 1))
+                ; Value.Array((Type.INT, Type.INT, [ (Value.Int 2, Value.Int 40) ;(Value.Int 6, Value.Int 20)],
+                                                     Value.Int 0))
+                ; Value.Array((Type.INT, Type.INT, [ (Value.Int 3 , Value.Int 6); (Value.Int 1, Value.Int 20)], 
+                                                      Value.Int 30))|];
+    constants = []
+  } in Alcotest.(check string) "identical" "(store a k v)" result.string
```

Comment:
You should also test storing to a location that already has a value, and similarly selecting from that location to make sure that we always fetch the updated value.

Reply:
To clarify, in the comment above I was suggesting you to add one more input `(a, k, v)` such that `a` already has `k` . So we want to test if the synthesizer can learn to override that index `k` by storing the new `v` there. In all the existing inputs the `k` is fresh (not stored in the `a`). 

Note that this is different from what `select_sem_test` does -- it checks that `select` and `store` are compatible.

Reply:
I have added the following for store, is this what you were looking for? (Also, I have added a similar test for ```select```) 
```let select_test_dup_key () =
  let open Synthesizer in
  let result = solve ~config:{ Config.default with logic = Logic.of_string "ALIA" } {
    arg_names = [ "a" ; "k" ];
    inputs = [ (Array.map ~f:(fun (a,b,c,d) -> Value.Array (a,b,c,d))
                     [| (Type.INT, Type.INT, [ (Value.Int 3, Value.Int 10);(Value.Int 3, Value.Int 30); (Value.Int 2, Value.Int 20); (Value.Int 1, Value.Int 10) ], Value.Int 1)
                     ; (Type.INT, Type.INT, [ (Value.Int 2, Value.Int 20); (Value.Int 1, Value.Int 1024) ], Value.Int 0)
                     ; (Type.INT, Type.INT, [(Value.Int 0 , Value.Int 0)], Value.Int 30) |])
             ; [| Value.Int 3 ; Value.Int 2 ; Value.Int 3 |] ];
    outputs = [| Value.Int 10 ; Value.Int 20 ; Value.Int 30 |];
    constants = []
  } in Alcotest.(check string) "identical" "(select a k)" result.string```